### PR TITLE
Remove py2 support from urls.py

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY

Remove py2 support from urls.py

##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: test/integration/targets/unarchive/tasks/test_download.yml [hunk 1]
```diff
@@ -6,19 +6,9 @@
...
-        - ndg_httpsclient==0.4.4
-        - pyOpenSSL==16.2.0
-      state: present
-    when:
-      - ansible_facts.distribution == 'CentOS'
-      - not ansible_facts.python.has_sslcontext
   - name: unarchive a tar from an URL
     unarchive:
-      src: "https://releases.ansible.com/ansible/ansible-latest.tar.gz"
+      src: "https://ftp.gnu.org/gnu/hello/hello-1.3.tar.gz"
```
Comment (jborean93) [inline test/integration/targets/unarchive/tasks/test_download.yml]: Should we look at using one of the CI .tar.gz files in Ansible Core CI S3 bucket, for example `https://ansible-ci-files.s3.amazonaws.com/test/integration/targets/apt/echo-hello-source.tar.gz`. We can also upload this `.tar.gz` if it's legally possible.

File: lib/ansible/module_utils/urls.py [hunk 2]
```diff
@@ -831,175 +386,111 @@ def missing_gzip_error():
...
+            data=data,
+            headers=req_headers,
+            origin_req_host=origin_req_host,
+            unverifiable=True,
+            method=method.upper(),
+        )
 
 
-def make_context(cafile=None, cadata=None, ciphers=None, validate_certs=True, client_cert=None, client_key=None):
+def make_context(cafile=None, cadata=None, capath=None, ciphers=None, validate_certs=True, client_cert=None,
```
Comment (jborean93) [inline lib/ansible/module_utils/urls.py:471]: Are we ok with this potentially breaking calls with positional args? It's not a `_` method but theoretically others could be using this outside of the util.

File: test/integration/targets/unarchive/tasks/test_download.yml (hunk 1 above)
Comment (sivel) [reply test/integration/targets/unarchive/tasks/test_download.yml]: I will look into using the existing file. I couldn't seem to find one already used by existing tests.

File: lib/ansible/module_utils/urls.py (hunk 2 above)
Comment (sivel) [reply lib/ansible/module_utils/urls.py:471]: I did check all collections from galaxy, and none were using it. But I'll double check, and if necessary reorder things.

File: test/integration/targets/unarchive/tasks/test_download.yml (hunk 1 above)
Comment (sivel) [reply test/integration/targets/unarchive/tasks/test_download.yml]: Done

**Review by jborean93 — APPROVED** [review]

**Pull request merged**